protocol/communities/persistence.go

```diff
@@ -1872,3 +1873,148 @@ func (p *Persistence) UpsertAppliedCommunityEvents(communityID types.HexBytes, p
 	}
 	return err
 }
+
+func (p *Persistence) InvalidateDecryptedCommunityCacheForKeys(keys []*encryption.HashRatchetInfo) error {
+	tx, err := p.db.BeginTx(context.Background(), &sql.TxOptions{})
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		if err == nil {
+			err = tx.Commit()
+			return
+		}
+		// don't shadow original error
+		_ = tx.Rollback()
+	}()
+
+	if len(keys) == 0 {
+		return nil
+	}
+	idsArgs := make([]interface{}, 0, len(keys))
+	for _, k := range keys {
+		idsArgs = append(idsArgs, k.KeyID)
+	}
+
+	inVector := strings.Repeat("?, ", len(keys)-1) + "?"
+
+	query := "SELECT DISTINCT(community_id) FROM encrypted_community_description_missing_keys WHERE key_id IN (" + inVector + ")" // nolint: gosec
+
+	var communityIDs []interface{}
+	rows, err := tx.Query(query, idsArgs...)
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var communityID []byte
+		err = rows.Scan(&communityID)
+		if err != nil {
+			return err
+		}
+		communityIDs = append(communityIDs, communityID)
+	}
+
+	query = "DELETE FROM encrypted_community_description_cache WHERE id IN (" + inVector + ")" //nolint: gosec
```

Comment:
yes, the cache should be fully invalidated if any of the missing keys is received, so we should remove the cached description, since we should be able to decrypt more data from it